### PR TITLE
add _calcTotalAmount

### PR DESCRIPTION
**Pull request by Lomet**
It seems that when we do Last Id, it works better on the create (sum only 1 time)
but we are getting complex logic on the split.

when we do the First Id, we need to loop 2 times.

### CODE REVIEW

File: contracts/LockDealBundleProvider/LockDealBundleProvider.sol [hunk 1]
```diff
@@ -35,29 +35,18 @@ contract LockDealBundleProvider is
         require(providerCount == providerParams.length, "providers and params length mismatch");
         require(providerCount > 1, "providers length must be greater than 1");
 
-        // create a new bundle pool owned by the owner with `0` token trasnfer amount
-        poolId = lockDealNFT.mint(owner, token, msg.sender, 0, address(this));
+        uint256 totalAmount = _calcTotalAmount(providerParams);
```
Comment (0xSolcroot) [inline contracts/LockDealBundleProvider/LockDealBundleProvider.sol:38]: Hmm... We iterate the same `for` statement twice. Is it fine?

File: contracts/LockDealBundleProvider/LockDealBundleProvider.sol [hunk 2]
```diff
@@ -35,29 +35,18 @@ contract LockDealBundleProvider is
...
-            if (i == providerCount - 1) {
-                // in case of the last sub pool, mint the NFT owned by the BunderDealProvider with `totalStartAmount` token transfer amount
-                // set the bundle provider state with the last sub pool Id
-                bundlePoolIdToLastSubPoolId[poolId] = _createNewSubPool(address(this), token, msg.sender, totalStartAmount, provider, params);
-            } else {
-               // mint the NFT owned by the BunderDealProvider with 0 token transfer amount
-                _createNewSubPool(address(this), token, msg.sender, 0, provider, params);
-            }
+            // mint the NFT owned by the BunderDealProvider with 0 token transfer amount
+            _createNewSubPool(address(this), token, msg.sender, 0, provider, params);
```
Comment (0xSolcroot) [inline contracts/LockDealBundleProvider/LockDealBundleProvider.sol:49]: Missing `bundlePoolIdToLastSubPoolId` storage set

**Review by 0xSolcroot — COMMENTED** [review]
Looks good!
Just a a few small stuff to discuss.

File: contracts/LockDealBundleProvider/LockDealBundleProvider.sol (hunk 2 above)
Comment (Lomet) [reply contracts/LockDealBundleProvider/LockDealBundleProvider.sol:49]: this needs to be made with _register.

File: contracts/LockDealBundleProvider/LockDealBundleProvider.sol (hunk 1 above)
Comment (Lomet) [reply contracts/LockDealBundleProvider/LockDealBundleProvider.sol:38]: we need to know the sum when opening the main pool. I think its fine here.

File: contracts/LockDealBundleProvider/LockDealBundleProvider.sol (hunk 2 above)
Comment (0xSolcroot) [reply contracts/LockDealBundleProvider/LockDealBundleProvider.sol:49]: Moving to the issue #140

**Pull request merged**